### PR TITLE
Adding info and changes to cromwell json template.

### LINKED ISSUE #603: Fix comments in wdl template json file
Comments are not supported in json.  These need to be removed.

### PR DESCRIPTION
**Pull request by LeeTL1220**
Made templates more like python templates.

Closes #603

### CODE REVIEW

File: scripts/wdl/README.md [hunk 1]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+
+To get started, copy the relevant ``*.template.json`` for the workflow you wish to run and adjust parameters accordingly.  This file has reasonable default parameters for exome or targeted panels.  
+- For WGS, please see notes below for ``isWGS``, ``wgsBinSize``, ``seg_param_undoSplits``, and ``seg_param_undoSD``
+- Values starting with ``$__`` *must* be replaced with values for your run.  Please note that python [templates](https://docs.python.org/2/library/string.html#template-strings) can be useful for replacing these values
+
+
+#### Explanation of fields in case workflow
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples
```
Comment (samuelklee) [inline scripts/wdl/README.md]: period at end of sentence

File: scripts/wdl/case_gatk_acnv_workflow_template.json [hunk 2]
```diff
@@ -1,36 +1,35 @@
-/* User needs to populate 'FILL_ME_IN' and save with new filename. */
 {
-  "case_gatk_acnv_workflow.input_bam_list": "FILL_ME_IN",  /* tsv file consisting of corresponding bam files and entity names as described in case_gatk_acnv_workflow.wdl  */
+  "case_gatk_acnv_workflow.input_bam_list": "$__input_bam_list___",
```
Comment (samuelklee) [inline scripts/wdl/case_gatk_acnv_workflow_template.json]: extra trailing underscore?

File: scripts/wdl/pon_gatk_workflow_template.json [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
-  "pon_gatk_workflow.target_file": "FILL_ME_IN", /* same file as was used for the PoN */
-  "pon_gatk_workflow.gatk_jar": "FILL_ME_IN", /* absolute path to gatk-protected.jar */
+  "pon_gatk_workflow.isWGS": "false", 
+  "pon_gatk_workflow.max_open_files": "1000",
+  "pon_gatk_workflow.wgsBinSize": "10000",  
+  "pon_gatk_workflow.enable_gc_correction": "true", 
+  "pon_gatk_workflow.ref_fasta": "$__ref_fasta__",
+  "pon_gatk_workflow.ref_fasta_fai": "$__ref_fasta_fai__",
+  "pon_gatk_workflow.pon_entity_id": "$__output_pon_name__",
+  "pon_gatk_workflow.normal_bam_list": "$__input_normal_bam_list___",
```
Comment (samuelklee) [inline scripts/wdl/pon_gatk_workflow_template.json]: extra trailing underscore?

File: scripts/wdl/README.md [hunk 4]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.seg_param_kmax`` -- (Advanced)  Segmenter parameter, default=25 
+- ``case_gatk_acnv_workflow.seg_param_nmin`` -- (Advanced)  Segmenter parameter, default=200 
+- ``case_gatk_acnv_workflow.seg_param_eta`` -- (Advanced)  Segmenter parameter, default=0.05 
+- ``case_gatk_acnv_workflow.seg_param_trim`` -- (Advanced)  Segmenter parameter, default=0.025 
+- ``case_gatk_acnv_workflow.seg_param_undoSplits`` -- Segmenter parameter, default=NONE   **For WGS: SDUNDO**
+- ``case_gatk_acnv_workflow.seg_param_undoPrune`` -- (Advanced)  Segmenter parameter, default=0.05 
+- ``case_gatk_acnv_workflow.seg_param_undoSD`` -- Segmenter parameter, default=3 **For WGS: 2**
+
+
+#### Explanation of fields in PoN workflow
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Why not put the PoN workflow before the case workflow?

File: scripts/wdl/README.md [hunk 5]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples.
+
+- ``pon_gatk_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true",
```
Comment (samuelklee) [inline scripts/wdl/README.md]: description missing

File: scripts/wdl/README.md [hunk 6]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``pon_gatk_workflow.enable_gc_correction`` --  "true" is recommended,  true/false option change to false to bypass gc bias correction step.  All case samples must be run with the same value as used in PoN creation.
+- ``pon_gatk_workflow.ref_fasta`` -- Path to reference fasta file. Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta
+- ``pon_gatk_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai
+- ``pon_gatk_workflow.pon_entity_id`` --  Name of the final PoN file
+- ``pon_gatk_workflow.normal_bam_list`` --  tsv file consisting of corresponding bam and corresponding index files as described in pon_gatk_workflow.wdl
+- ``pon_gatk_workflow.ref_fasta_dict`` --  Path to reference dict file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.dict
+- ``pon_gatk_workflow.target_file`` --  Target file (NOT in bed format) that was used to describe the baits in capture (exome) samples.
+- ``pon_gatk_workflow.gatk_jar`` -- absolute path to gatk-protected.jar
```
Comment (samuelklee) [inline scripts/wdl/README.md]: capitalize Absolute to be consistent

File: scripts/wdl/README.md [hunk 7]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``pon_gatk_workflow.enable_gc_correction`` --  "true" is recommended,  true/false option change to false to bypass gc bias correction step.  All case samples must be run with the same value as used in PoN creation.
+- ``pon_gatk_workflow.ref_fasta`` -- Path to reference fasta file. Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta
+- ``pon_gatk_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai
+- ``pon_gatk_workflow.pon_entity_id`` --  Name of the final PoN file
+- ``pon_gatk_workflow.normal_bam_list`` --  tsv file consisting of corresponding bam and corresponding index files as described in pon_gatk_workflow.wdl
+- ``pon_gatk_workflow.ref_fasta_dict`` --  Path to reference dict file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.dict
+- ``pon_gatk_workflow.target_file`` --  Target file (NOT in bed format) that was used to describe the baits in capture (exome) samples.
```
Comment (samuelklee) [inline scripts/wdl/README.md]: I would say something here about using `ConvertBedToTargetFile` if their target file is a bed file.

File: scripts/wdl/README.md [hunk 8]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+#### Explanation of fields in case workflow
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples
+
+- ``case_gatk_acnv_workflow.input_bam_list`` -- tsv file consisting of corresponding bam files and entity names as described in case_gatk_acnv_workflow.wdl
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.grouping`` -- (Advanced) Always choose "SAMPLE",
+- ``case_gatk_acnv_workflow.common_snp_list`` --  SNP list to use in het pulldown 
+- ``case_gatk_acnv_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai 
+- ``case_gatk_acnv_workflow.target_file`` --  same target file (NOT bed format) as was used for the PoN creation
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Capitalize Same.  Also see comment below about `ConvertBedToTargetFile`.

File: scripts/wdl/README.md [hunk 9]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+
+
+#### Explanation of fields in case workflow
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples
+
+- ``case_gatk_acnv_workflow.input_bam_list`` -- tsv file consisting of corresponding bam files and entity names as described in case_gatk_acnv_workflow.wdl
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.grouping`` -- (Advanced) Always choose "SAMPLE",
+- ``case_gatk_acnv_workflow.common_snp_list`` --  SNP list to use in het pulldown
```
Comment (samuelklee) [inline scripts/wdl/README.md]: in het pulldown -> for GetBayesianHetCoverage

File: scripts/wdl/README.md [hunk 10]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- Values starting with ``$__`` *must* be replaced with values for your run.  Please note that python [templates](https://docs.python.org/2/library/string.html#template-strings) can be useful for replacing these values
+
+
+#### Explanation of fields in case workflow
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples
+
+- ``case_gatk_acnv_workflow.input_bam_list`` -- tsv file consisting of corresponding bam files and entity names as described in case_gatk_acnv_workflow.wdl
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.grouping`` -- (Advanced) Always choose "SAMPLE",
```
Comment (samuelklee) [inline scripts/wdl/README.md]: trailing comma?

File: scripts/wdl/README.md [hunk 11]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.common_snp_list`` --  SNP list to use in het pulldown 
+- ``case_gatk_acnv_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai 
+- ``case_gatk_acnv_workflow.target_file`` --  same target file (NOT bed format) as was used for the PoN creation
+- ``case_gatk_acnv_workflow.gatk_jar`` --  absolute path to gatk-protected.jar
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Capitalize Absolute.

File: scripts/wdl/README.md [hunk 12]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples
+
+- ``case_gatk_acnv_workflow.input_bam_list`` -- tsv file consisting of corresponding bam files and entity names as described in case_gatk_acnv_workflow.wdl
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.grouping`` -- (Advanced) Always choose "SAMPLE",
+- ``case_gatk_acnv_workflow.common_snp_list`` --  SNP list to use in het pulldown 
+- ``case_gatk_acnv_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai 
+- ``case_gatk_acnv_workflow.target_file`` --  same target file (NOT bed format) as was used for the PoN creation
+- ``case_gatk_acnv_workflow.gatk_jar`` --  absolute path to gatk-protected.jar 
+- ``case_gatk_acnv_workflow.PadTargets.mem`` -- (Advanced) Number of GB RAM to use when running PadTargets.  Always use 1, unless your target file is really large
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Nitpicking, but I would be consistent about whether or not to end all bullet points with a period.

File: scripts/wdl/README.md [hunk 13]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``pon_gatk_workflow.isWGS`` -- Change to "true"" if running WGS samples
+- ``pon_gatk_workflow.max_open_files`` -- (Advanced) "100",  Maximum number of files to combine simultaneously while combining read counts 
+- ``pon_gatk_workflow.wgsBinSize`` --  Size of bins (in bp) for WGS read counts CLI.  *This must be the same values used for all case samples*.  Ignored if not running WGS.
+- ``pon_gatk_workflow.enable_gc_correction`` --  "true" is recommended,  true/false option change to false to bypass gc bias correction step.  All case samples must be run with the same value as used in PoN creation.
+- ``pon_gatk_workflow.ref_fasta`` -- Path to reference fasta file. Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta
+- ``pon_gatk_workflow.ref_fasta_fai`` --  Path to reference fasta fai file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta.fai
+- ``pon_gatk_workflow.pon_entity_id`` --  Name of the final PoN file
+- ``pon_gatk_workflow.normal_bam_list`` --  tsv file consisting of corresponding bam and corresponding index files as described in pon_gatk_workflow.wdl
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Maybe just put the full path to pon_gatk_workflow.wdl here?

File: scripts/wdl/README.md [hunk 14]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true" 
+- ``case_gatk_acnv_workflow.plots_dir`` -- (Advanced)  path to store produced plots for CopyRatio and ACNV 
+- ``case_gatk_acnv_workflow.call_cnloh_dir`` -- (Advanced)  path to directory to store CNLoH and splits results 
+- ``case_gatk_acnv_workflow.enable_gc_correction`` -- "true" is recommended. true/false option change to false to bypass gc bias correction step 
+- ``case_gatk_acnv_workflow.isWGS`` -- Change to true if running WGS samples.  You should also change ``case_gatk_acnv_workflow.seg_param_undoSplits`` and ``case_gatk_acnv_workflow.seg_param_undoSD`` to reduce (not eliminate) hyper-segmentation
+- ``case_gatk_acnv_workflow.wgsBinSize`` -- "10000" .  Size of bins (in bp) for WGS read counts CLI.  *This must be the same as was used to generate the PoN*.  Ignored if not running WGS.
+
+*For segmenter parameters, see the explanation in the Segmenter tool or page 11 of https://bioconductor.org/packages/release/bioc/manuals/DNAcopy/man/DNAcopy.pdf*
```
Comment (samuelklee) [inline scripts/wdl/README.md]: Segmenter -> PerformSegmentation here and below.  However, the argument descriptions there just refer to the manual, so I would just say "see pgs. 11-12 of https://bioconductor.org/packages/release/bioc/manuals/DNAcopy/man/DNAcopy.pdf"

File: scripts/wdl/README.md [hunk 15]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.NormalizeSomaticReadCounts.mem`` -- "4" Number of GB RAM to use when running 
+- ``case_gatk_acnv_workflow.PadTargets.pd`` -- (Advanced) "250" has been shown to improve results.  This parameter is number of bases to pad the target file in ``PadTargets``
+- ``case_gatk_acnv_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true" 
+- ``case_gatk_acnv_workflow.plots_dir`` -- (Advanced)  path to store produced plots for CopyRatio and ACNV 
+- ``case_gatk_acnv_workflow.call_cnloh_dir`` -- (Advanced)  path to directory to store CNLoH and splits results 
+- ``case_gatk_acnv_workflow.enable_gc_correction`` -- "true" is recommended. true/false option change to false to bypass gc bias correction step 
+- ``case_gatk_acnv_workflow.isWGS`` -- Change to true if running WGS samples.  You should also change ``case_gatk_acnv_workflow.seg_param_undoSplits`` and ``case_gatk_acnv_workflow.seg_param_undoSD`` to reduce (not eliminate) hyper-segmentation
+- ``case_gatk_acnv_workflow.wgsBinSize`` -- "10000" .  Size of bins (in bp) for WGS read counts CLI.  *This must be the same as was used to generate the PoN*.  Ignored if not running WGS.
```
Comment (samuelklee) [inline scripts/wdl/README.md]: extra whitespace

File: scripts/wdl/README.md [hunk 16]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.ref_fasta_dict`` --  Path to reference dict file.  Broad internal: /seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.dict 
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.keepduplicatereads`` -- (Advanced) Whether coverage calculations should count duplicate reads.  Setting this to "true" has been shown (empirically) to improve performance.  "false" is also supported.
+- ``case_gatk_acnv_workflow.PoN`` --  Path to GATK PoN file
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.transform `` -- (Advanced) Always "PCOV"
+- ``case_gatk_acnv_workflow.ref_fasta`` -- Path to reference fasta file.  Broad internal: ``/seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta`` 
+- ``case_gatk_acnv_workflow.NormalizeSomaticReadCounts.mem`` -- "4" Number of GB RAM to use when running 
+- ``case_gatk_acnv_workflow.PadTargets.pd`` -- (Advanced) "250" has been shown to improve results.  This parameter is number of bases to pad the target file in ``PadTargets``
+- ``case_gatk_acnv_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true"
```
Comment (samuelklee) [inline scripts/wdl/README.md]: description?

File: scripts/wdl/README.md [hunk 17]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+- ``case_gatk_acnv_workflow.PoN`` --  Path to GATK PoN file
+- ``case_gatk_acnv_workflow.CalculateTargetCoverage.transform `` -- (Advanced) Always "PCOV"
+- ``case_gatk_acnv_workflow.ref_fasta`` -- Path to reference fasta file.  Broad internal: ``/seq/references/Homo_sapiens_assembly19/v1/Homo_sapiens_assembly19.fasta`` 
+- ``case_gatk_acnv_workflow.NormalizeSomaticReadCounts.mem`` -- "4" Number of GB RAM to use when running 
+- ``case_gatk_acnv_workflow.PadTargets.pd`` -- (Advanced) "250" has been shown to improve results.  This parameter is number of bases to pad the target file in ``PadTargets``
+- ``case_gatk_acnv_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true" 
+- ``case_gatk_acnv_workflow.plots_dir`` -- (Advanced)  path to store produced plots for CopyRatio and ACNV 
+- ``case_gatk_acnv_workflow.call_cnloh_dir`` -- (Advanced)  path to directory to store CNLoH and splits results 
+- ``case_gatk_acnv_workflow.enable_gc_correction`` -- "true" is recommended. true/false option change to false to bypass gc bias correction step
```
Comment (samuelklee) [inline scripts/wdl/README.md]: gc -> GC

File: scripts/wdl/README.md [hunk 18]
```diff
@@ -6,5 +6,70 @@ https://github.com/broadinstitute/wdl
...
+#### Explanation of fields in PoN workflow
+
+Fields marked as Advanced almost never require a user to adjust the default value.
+Reference used must be the same between PoN and case samples.
+
+- ``pon_gatk_workflow.disable_sequence_dictionary_validation`` -- (Advanced) "true",
+- ``pon_gatk_workflow.isWGS`` -- Change to "true"" if running WGS samples
+- ``pon_gatk_workflow.max_open_files`` -- (Advanced) "100",  Maximum number of files to combine simultaneously while combining read counts 
+- ``pon_gatk_workflow.wgsBinSize`` --  Size of bins (in bp) for WGS read counts CLI.  *This must be the same values used for all case samples*.  Ignored if not running WGS.
+- ``pon_gatk_workflow.enable_gc_correction`` --  "true" is recommended,  true/false option change to false to bypass gc bias correction step.  All case samples must be run with the same value as used in PoN creation.
```
Comment (samuelklee) [inline scripts/wdl/README.md]: gc -> GC

**Comment by samuelklee** [comment]
Just some style/formatting issues.  Thanks for putting this together, back to you @LeeTL1220 !

File: scripts/wdl/README.md (hunk 1 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/case_gatk_acnv_workflow_template.json (hunk 2 above)
Comment (LeeTL1220) [inline scripts/wdl/case_gatk_acnv_workflow_template.json]: Done

File: scripts/wdl/pon_gatk_workflow_template.json (hunk 3 above)
Comment (LeeTL1220) [inline scripts/wdl/pon_gatk_workflow_template.json]: Done

File: scripts/wdl/README.md (hunk 4 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 5 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 6 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 7 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Good catch!  Done

File: scripts/wdl/README.md (hunk 8 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Dome

File: scripts/wdl/README.md (hunk 9 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 10 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 11 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 12 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Tried to make all capital.

File: scripts/wdl/README.md (hunk 14 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 15 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: @samuelklee Where?

File: scripts/wdl/README.md (hunk 16 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 17 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

File: scripts/wdl/README.md (hunk 18 above)
Comment (LeeTL1220) [inline scripts/wdl/README.md]: Done

**Comment by LeeTL1220** [comment]
@samuelklee One minor thing that I could not find.

File: scripts/wdl/README.md (hunk 15 above)
Comment (samuelklee) [inline scripts/wdl/README.md]: "10000" .

Comment (samuelklee) [inline scripts/wdl/README.md]: looks like you filled it in already!

**Comment by samuelklee** [comment]
looks good to squash and merge!

**Pull request merged**